### PR TITLE
small fixes/improvements for group protocol

### PR DESCRIPTION
**Pull request by ant9000**
The most notable difference: participants added to ListGroupsResultIqProtocolEntity; other stuff is mostly cosmetic.

### CODE REVIEW

File: yowsup/layers/network/layer.py [hunk 1]
```diff
@@ -56,7 +56,9 @@ def handle_connect(self):
 
     def handle_close(self, reason = "Connection Closed"):
         logger.debug("Disconnected, reason: %s" % reason)
-        self.emitEvent(YowLayerEvent(self.__class__.EVENT_STATE_DISCONNECTED, reason = reason, detached=True))
+        ev = YowLayerEvent(self.__class__.EVENT_STATE_DISCONNECTED, reason = reason, detached=True)
+        self.emitEvent(ev)
+        self.getStack().broadcastEvent(ev)
```
Comment (tgalal) [inline yowsup/layers/network/layer.py]: why is that ? You are sending the event twice this way, bottom-up, and up-bottom

**Comment by tgalal** [comment]
Thanks!

File: yowsup/layers/network/layer.py (hunk 1 above)
Comment (ant9000) [inline yowsup/layers/network/layer.py]: Because with your version the event never reached my topmost layer, where I need it to try to reconnect.

**Comment by ant9000** [comment]
On Sun, Jul 19, 2015 at 1:14 AM, Tarek <notifications@github.com> wrote:
>
> Thanks!

You're welcome!

Antonio

**Comment by tgalal** [comment]
I'm not sure why this is still open I believe I merged the group fixes

**Comment by sguron** [comment]
@tgalal Are you sure ? I dont see these changes in the current source.

https://github.com/tgalal/yowsup/blob/master/yowsup/layers/network/layer.py#L70

**Pull request closed without merging**